examples/chip-tool/commands/payload/ParseCommand.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+ *   Copyright (c) 2020 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+#include "ParseCommand.h"
+#include <setup_payload/QRCodeSetupPayloadParser.h>
+#include <setup_payload/ManualSetupPayloadParser.h>
+#include <setup_payload/SetupPayload.h>
+#include <stdio.h>
+#include <iostream>
+#include <stdio.h>
+#include <support/verhoeff/Verhoeff.h>
+
+using namespace ::chip;
+using namespace ::chip::DeviceController;
+
+CHIP_ERROR ParseCommand::Run(ChipDeviceController * dc, NodeId remoteId)
+{    
+    std::string codeString(mCode);
+    SetupPayload payload;
+    
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    err = Parse(codeString, payload);
+
+    SuccessOrExit(err);
+    
+    Print(payload);
```

Comment:
In one of the following commands I'm asking if you can update the prototype of `void Print` to be `CHIP_ERROR Print`. Because of this the code can be rewritten such as:
```
CHIP_ERROR ParseCommand::Run(ChipDeviceController * dc, NodeId remoteId)
{
    CHIP_ERROR err = CHIP_NO_ERROR;
    SetupPayload payload;

    err = Parse(mCode, payload);
    SuccessOrExit(err);

    err = Print(payload);
    SuccessOrExit(err);

exit:
    return err;
}
```

Reply:
Addressed in the new revision.